trinity/tx_pool/pool.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from typing import (
+    cast,
+    Iterable,
+    List
+)
+import uuid
+
+from bloom_filter import (
+    BloomFilter
+)
+
+from evm.rlp.transactions import (
+    BaseTransactionFields
+)
+from p2p.eth import (
+    Transactions
+)
+from p2p.peer import (
+    BasePeer,
+    ETHPeer,
+    PeerPool,
+    PeerPoolSubscriber,
+)
+from p2p.service import (
+    BaseService
+)
+
+
+class TxPool(BaseService, PeerPoolSubscriber):
+    """
+    The :class:`~trinity.tx_pool.pool.TxPool` class is responsible for holding and relaying
+    of transactions, represented as :class:`~evm.rlp.transactions.BaseTransaction` among the
+    connected peers.
+
+      .. note::
+
+        This is a minimal vialble implementation that only relays transactions but doesn't actually
+        hold on to them yet. It's still missing many features of a grown up transaction pool.
+    """
+    logger = logging.getLogger("trinity.tx_pool.TxPool")
+
+    def __init__(self, peer_pool: PeerPool) -> None:
+        super().__init__()
+        self._peer_pool = peer_pool
+        # 1m should give us 9000 blocks before that filter becomes less reliable
+        # It should take up about 1mb of memory
+        self._bloom = BloomFilter(max_elements=1000000)
+        self._bloom_salt = str(uuid.uuid4())
+
+    def register_peer(self, peer: BasePeer) -> None:
+        pass
+
+    async def _run(self) -> None:
+        self.logger.info("Running Tx Pool")
+
+        with self.subscribe(self._peer_pool):
+            while True:
+                peer: ETHPeer
+                peer, cmd, msg = await self.wait(
+                    self.msg_queue.get(), token=self.cancel_token)
+
+                if isinstance(cmd, Transactions):
+                    await self._handle_tx(peer, msg)
+
+    async def _handle_tx(self, peer: ETHPeer, txs: List[BaseTransactionFields]) -> None:
+
+        self.logger.debug('Received transactions from %r: %r', peer, txs)
+
+        self._add_txs_to_bloom(peer, txs)
+
+        for receiving_peer in self._peer_pool.peers:
+            receiving_peer = cast(ETHPeer, receiving_peer)
+            if receiving_peer is not peer:
+                filtered_tx = self._filter_tx_for_peer(receiving_peer, txs)
+                if len(filtered_tx) > 0:
+                    self.logger.debug(
+                        'Sending transactions to %r: %r',
+                        receiving_peer,
+                        filtered_tx
+                    )
+                    receiving_peer.sub_proto.send_transactions(filtered_tx)
+                    self._add_txs_to_bloom(receiving_peer, filtered_tx)
+
+    def _filter_tx_for_peer(
+            self,
+            peer: BasePeer,
+            txs: List[BaseTransactionFields]) -> List[BaseTransactionFields]:
+
+        return [val for val in txs if self._construct_bloom_entry(peer, val) not in self._bloom]
```

Comment:
*nitpick*

I find these easier to read as

```python
return [
    val 
    for val 
    in txs 
    if self._construct_bloom_entry(peer, val) not in self._bloom
]
# or if you're more into brevity

return [
    val for val in txs 
    if self._construct_bloom_entry(peer, val) not in self._bloom
]
```

Reply:
yep, same here, I'm more used to 

```python
txs.filter(lambda tx: return self._construct_bloom_entry(peer, tx) not in self._bloom)
```

I like the second of your suggestions more than the first.